### PR TITLE
CNV-11198: Added info about MAC spoof check

### PR DESCRIPTION
**Pull request by sjhala-ccs**
This PR addresses [CNV-11198](https://issues.redhat.com/browse/CNV-11198)

Updated existing assembly "Attaching a VM to multiple networks" to add content about the MAC spoof check feature

Applies to enterprise-4.9

Preview: https://deploy-preview-35806--osdocs.netlify.app/openshift-enterprise/latest/virt/virtual_machines/vm_networking/virt-attaching-vm-multiple-networks?utm_source=github&utm_campaign=bot_dp#virt-creating-bridge-nad-cli_virt-attaching-multiple-networks

### CODE REVIEW

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 1]
```diff
@@ -46,10 +48,11 @@ spec:
 <3> The name for the configuration. It is recommended to match the configuration name to the `name` value of the network attachment definition.
 <4> The actual name of the Container Network Interface (CNI) plug-in that provides the network for this network attachment definition. Do not change this field unless you want to use a different CNI.
 <5> The name of the Linux bridge configured on the node.
-<6> Optional: The VLAN tag.
-<7> The CNI plug-in that allows the MAC pool manager to assign a unique MAC address to the connection.
+<6> The MAC spoof check. When set to `true`, this feature provides security against a MAC spoofing attack where an intruder impersonates a device with a valid MAC address to bypass access control lists and breach the network.
```
Comment (EdDev) [inline modules/virt-creating-bridge-nad-cli.adoc:51]: I would prefer to see a description of what it does and not why it should be used. At least, the first part if not clear and the second is to accurate.

This option provides security against a MAC spoofing attack by allowing only a single MAC address (which cannot be changed ) to exit the pod.

(if the user will try to change the MAC, traffic will be blocked)

I think the second part is not accurate because this is not protecting against an external intruder that may attack the pod from outside. So IMO just describing what it does should be enough.

Comment (sjhala-ccs) [reply modules/virt-creating-bridge-nad-cli.adoc:51]: Reworded per your suggestion.

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 2]
```diff
@@ -46,10 +48,11 @@ spec:
 <3> The name for the configuration. It is recommended to match the configuration name to the `name` value of the network attachment definition.
 <4> The actual name of the Container Network Interface (CNI) plug-in that provides the network for this network attachment definition. Do not change this field unless you want to use a different CNI.
 <5> The name of the Linux bridge configured on the node.
-<6> Optional: The VLAN tag.
-<7> The CNI plug-in that allows the MAC pool manager to assign a unique MAC address to the connection.
+<6> Optional: Flag to enable MAC spoof check. When set to `true`, you cannot change the MAC address of the pod or guest interface. This attribute provides security against a MAC spoofing attack by allowing only a single MAC address to exit the pod.
+<7> Optional: The VLAN tag.
+<8> The CNI plug-in that allows the MAC pool manager to assign a unique MAC address to the connection.
```
Comment (EdDev) [inline modules/virt-creating-bridge-nad-cli.adoc:53]: This is not directly related to this change, so it may be appropriate to change it in a following PR: We no longer require `cnv-tuning` to set the MAC address, cnv-bridge now supports it natively.

**Review by EdDev — APPROVED** [review]
Thank you, the text looks great now.
Please see a note inline about a different topic.

File: modules/virt-creating-bridge-nad-cli.adoc (hunk 2 above)
Comment (sjhala-ccs) [reply modules/virt-creating-bridge-nad-cli.adoc:53]: Maybe you can open a documentation bug to address that change?

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 3]
```diff
@@ -7,10 +7,11 @@
...
 
-[NOTE]
-====
-The network attachment definition must be in the same namespace as the pod or virtual machine.
-====
+.Prerequisites
+
+* The network attachment definition must be in the same namespace as the pod or virtual machine.
+
+* Optional: The node must support nftables and the `nft` binary must be deployed to enable MAC spoof check.
```
Comment (geetikakay) [inline modules/virt-creating-bridge-nad-cli.adoc:14]: I think for this to work , nftables are mandatory to have. Optional looks confusing.

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 4]
```diff
@@ -46,10 +48,11 @@ spec:
 <3> The name for the configuration. It is recommended to match the configuration name to the `name` value of the network attachment definition.
 <4> The actual name of the Container Network Interface (CNI) plug-in that provides the network for this network attachment definition. Do not change this field unless you want to use a different CNI.
 <5> The name of the Linux bridge configured on the node.
-<6> Optional: The VLAN tag.
-<7> The CNI plug-in that allows the MAC pool manager to assign a unique MAC address to the connection.
+<6> Optional: Flag to enable MAC spoof check. When set to `true`, you cannot change the MAC address of the pod or guest interface. This attribute provides security against a MAC spoofing attack by allowing only a single MAC address to exit the pod.
```
Comment (geetikakay) [inline modules/virt-creating-bridge-nad-cli.adoc:51]: one line if it's not set. If not set spoofing will not be checked.

File: modules/virt-creating-bridge-nad-cli.adoc [hunk 5]
```diff
@@ -33,10 +34,11 @@ spec:
       {
         "type": "cnv-bridge", <4>
         "bridge": "<bridge-interface>", <5>
-        "vlan": 1 <6>
+        "macspoofchk": true, <6>
+        "vlan": 1 <7>
       },
       {
-        "type": "cnv-tuning" <7>
+        "type": "cnv-tuning" <8>
```
Comment (geetikakay) [inline modules/virt-creating-bridge-nad-cli.adoc:41]: I have not seen this in ipsteam document. Is it very specific to downstream? plus how this and KMP work together. This also look for uniqueness of MAC and KMP does the same as well.

**Review by geetikakay — COMMENTED** [review]
QE review

File: modules/virt-creating-bridge-nad-cli.adoc (hunk 3 above)
Comment (sjhala-ccs) [reply modules/virt-creating-bridge-nad-cli.adoc:14]: Removed the word "Optional"

File: modules/virt-creating-bridge-nad-cli.adoc (hunk 4 above)
Comment (sjhala-ccs) [reply modules/virt-creating-bridge-nad-cli.adoc:51]: That is implied by the sentence "When set to true..."

File: modules/virt-creating-bridge-nad-cli.adoc (hunk 5 above)
Comment (sjhala-ccs) [reply modules/virt-creating-bridge-nad-cli.adoc:41]: Is this related to Edward's comment above?

File: modules/virt-creating-bridge-nad-cli.adoc (hunk 3 above)
Comment (EdDev) [reply modules/virt-creating-bridge-nad-cli.adoc]: Well, if no one uses mac-spoof-check, it is indeed optional. The sentence says that if one wants to use it, it is needed.
But I do not really mind if this gets removed.

File: modules/virt-creating-bridge-nad-cli.adoc (hunk 5 above)
Comment (EdDev) [reply modules/virt-creating-bridge-nad-cli.adoc:41]: This shows up on kubevirt user-guide:

> Note: For some CNI plugins without native support for custom MAC addresses, there is a workaround, which is to use the tuning CNI plugin to adjust pod interface MAC address. This can be used as follows:

**Review by bergerhoffer — COMMENTED** [review]
LGTM
